hazelcast-client/src/main/java/com/hazelcast/client/config/ClientProperties.java

```diff
@@ -22,6 +22,16 @@
 public class ClientProperties {
 
     /**
+     * Client shuffles the given member list to prevent all clients to connect to same node when
+     * this property is false. When set to true, client tries to connect to nodes in order as given.
+     */
```

Comment:
Perhaps it would be an idea to include a few lines why this property exist; so setting it to false helps to make tests deterministic. I don't know if it has any value for a production system.